src/analyze.ts

```diff
@@ -614,18 +619,57 @@ export default async function analyze(id: string, code: string, job: Job): Promi
               }
             break;
             case NODE_GYP_BUILD:
-              if (node.arguments.length === 1 && node.arguments[0].type === 'Identifier' &&
-                  node.arguments[0].name === '__dirname' && knownBindings.__dirname.shadowDepth === 0) {
+              // handle case: require("node-gyp-build")(__dirname)
+              const calledWithDirnameDirectly =
+                node.arguments.length === 1 &&
+                node.arguments[0].type === 'Identifier' &&
+                node.arguments[0].name === '__dirname';
+
+              // handle case: require("node-gyp-build")(path.join(__dirname, ".."))
+              const calledWithPathJoinDirname =
+                node.arguments.length === 1 &&
+                node.arguments[0].callee?.object?.name === "path" &&
+                node.arguments[0].callee?.property?.name === "join" &&
+                node.arguments[0].arguments.length === 2 &&
+                node.arguments[0].arguments[0].type === 'Identifier' &&
+                node.arguments[0].arguments[0].name === '__dirname' &&
+                node.arguments[0].arguments[1].type === 'Literal'
+
+              if ((calledWithDirnameDirectly || calledWithPathJoinDirname) &&
+                  knownBindings.__dirname.shadowDepth === 0) {
+
+                // Resolve the path in the case of passing path.join(__dirname, "..") instead of __dirname.
+                const pathJoinedDir =
+                  calledWithPathJoinDirname ? path.join(dir, node.arguments[0].arguments[1].value) : dir;
+
+                // zeromq.js uses @aminya's fork of node-gyp-build, consider it.
+                const validNodeGypBuildForkNames = ['node-gyp-build', '@aminya/node-gyp-build'];
+                if (node.callee.arguments[0].type !== 'Literal' ||
+                    !validNodeGypBuildForkNames.includes(node.callee.arguments[0].value)) {
+                  throw new Error("Unknown node-gyp-build fork name");
```

Comment:
We shouldn't throw. This will be impossible for users to comprehend.

Reply:
What would you say it's best then? Console error?

Reply:
Skip here and add a warning there: https://github.com/vercel/nft/pull/392#discussion_r1488419981

Reply:
Done.